content/articles/getting-started-with-unit-testing-with-spring-boot/index.md

```diff
@@ -0,0 +1,131 @@
+
+A new file `/src/main/test/java/controller/HelloControllerTest.java` will be created. Modify the file to have the code below:
+```java
+package controller;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+class HelloControllerTest {
+    @Test
+    void hello() {
+        HelloController controller = new HelloController(); // instance of the controller
+        String response = controller.hello("Hello world"); // act
+        Assertions.assertEquals("Hello world", response); // assert
+    }
+}
+```
+So far, we have created a test without including Spring Boot. Let us generate another test using the `JUnit 5` extension, which will have Spring extensions provided by Spring.  
+
+Repeat the same procedure to generate another test. Name the second test as `HelloControllerIntTest.java.` In `/src/main/test/java/controller/HelloControllerIntTest.java`, modify the code as shown below:
+```java
+package controller;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MockMvcResultHandlersDsl;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import static org.junit.jupiter.api.Assertions.*;
+@ExtendWith({SpringExtension.class})
+@WebMvcTest(HelloController.class)
+class HelloControllerIntTest {
+    @Autowired
+    private MockMvc mvc;
+    @Test
+    void hello() throws Exception {
+        RequestBuilder request = MockMvcRequestBuilders.get("/hello");
+        MvcResult result = mvc.perform(request).andReturn();
+        Assertions.assertEquals("Hello, Hello world", result.getResponse().getContentAsString());
+    }
+}
+```
+Let us look at the code snippet above:
+- `@ExtendWith` is an annotation used to extend the test with the SpringExtension.
+- `@WebMvcTest` - This annotation auto-configures the MockMVC (so we can auto-wire it as in the code above). We specify the class we want to test.
+
+Let us run our Controller to be sure it works correctly.
+
+> Note: If we use that structure to run the test, it will fail. Spring Boot will look for `@SpringBootConfiguration` at the package `/test/controller`, and it will not find it. To avoid this, let us redesign the directory structure. Move `/src/main/java/com.example.demo/DemoAplication.java` file to `/src/main/test/java/controller/`. Now, both the test and the application are in the same package.
```

Comment:
You don't want to put your tests and applications in the same package. I.e., the application should be in `/src/main/java/com.example.demo/DemoApplication.java` while the tests should be in `/src/main/test/java/com.example.demo`.